File: api/lib/reading_time.rb
module ReadingTime
    DEFAULT_WORDS_PER_MINUTE = 200

    def self.calculate(content, words_per_minute = DEFAULT_WORDS_PER_MINUTE)
        return 1 if content.nil? || content.strip.empty?

        words = content.split.size
        [(words / words_per_minute.to_f).ceil, 1].max
    end

    def self.format(content, words_per_minute = DEFAULT_WORDS_PER_MINUTE)
        minutes = calculate(content, words_per_minute)
        "#{minutes} #{'minute'.pluralize(minutes)} read"
    end
end

File: web/server/lib/reading-time.ts
const DEFAULT_WORDS_PER_MINUTE = 200;

const calculate = (content: string | null, wordsPerMinute = DEFAULT_WORDS_PER_MINUTE): number => {
  if (!content || content.trim().length === 0) return 1;

  const words = content.split(/\s+/).filter(Boolean).length;
  return Math.max(Math.ceil(words / wordsPerMinute), 1);
};

export const formatReadingTime = (content: string | null, wordsPerMinute = DEFAULT_WORDS_PER_MINUTE): string => {
  const minutes = calculate(content, wordsPerMinute);
  return `${minutes} ${minutes === 1 ? 'minute' : 'minutes'} read`;
};
